javascript/javascript1/week2/homework-js1-w2.js

```diff
@@ -0,0 +1,168 @@
+// freeCodeCamp username: yuejiaoshi
+// freeCodeCamp profile url: https://www.freecodecamp.org/yuejiaoshi
+
+//*****************************************/
+// Flight booking fullname function
+//*****************************************/
+function getFullname(firstname, surname, useFormalName = false, gender) {
+  if (!useFormalName) {
+    return firstname + " " + surname;
+  } else {
+    if (gender === "man") {
+      return "Lord" + " " + firstname + " " + surname;
+    } else if (gender === "woman") {
+      return "Lady" + " " + firstname + " " + surname;
+    } else {
+      return (
+        "Please provide gender of " +
+        firstname +
+        " " +
+        surname +
+        " for a formal name," +
+        " 'man' or 'woman'?"
+      );
+    }
+  }
+}
+const fullname1 = getFullname("Benjamin", "Hughes", true, "man");
+const fullname2 = getFullname("Yuejiao", "Shi", true, "woman");
+console.log(fullname1);
+console.log(fullname2);
+
+//*****************************************/
+// Event application
+//*****************************************/
+const currentDate = new Date();
+const todayIndex = currentDate.getDay();
+const weekdays = [
+  "Sunday",
+  "Monday",
+  "Tuesday",
+  "Wednesday",
+  "Thursday",
+  "Friday",
+  "Saturday",
+];
+
+function getEventWeekday(daysFromToday) {
+  const eventDay = weekdays[(todayIndex + daysFromToday) % 7];
+  return eventDay;
+}
+
+console.log(getEventWeekday(2));
+
+//*****************************************/
+// Weather wear
+//*****************************************/
+function getClothes(temp) {
+  if (temp <= -20) {
+    return "It's too cold, please do not go out.";
+  } else if (temp <= -5) {
+    return "Down jacket, warm scarf, and gloves";
+  } else if (temp <= 10) {
+    return "Heavier coat, sweater or hoodie, and jeans";
+  } else if (temp <= 20) {
+    return "Light jacket, light jeans";
+  } else if (temp <= 40) {
+    return "Skirt, shorts and a t-shirt";
+  } else if (40 < temp) {
+    return "It's too hot, please do not go out.";
+  }
+}
+
+const clothesToWear = getClothes(18);
+console.log(clothesToWear);
+
+//*****************************************/
+// Student manager
+//*****************************************/
+const class07Students = [];
+function addStudentToClass(studentName) {
+  // You write code here
+  const lowercaseClass07Students = class07Students.map((name) =>
```

Comment:
Is `lowercaseClass07Students` the best name once you remove the `toLowerCase()` method? Would something like `duplicateNames` be better to describe what's contained?